cmd/common/containerized-gadgets/build.go

```diff
@@ -0,0 +1,231 @@
+// Copyright 2023 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package containerizedgadgets
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/opencontainers/image-spec/specs-go"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/spf13/cobra"
+	"oras.land/oras-go/v2"
+	"oras.land/oras-go/v2/content"
+)
+
+type buildOptions struct {
+	progAmd64FilePath  string
+	progArm64FilePath  string
+	definitionFilePath string
+	tag                string
+}
+
+func NewBuildCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "build tag",
+		Short: "build a containerized gadget image",
+		RunE:  runBuild,
+	}
```

Comment:
I think all these commands should set `SilenceUsage: true` to avoid printing misleading (not useful) information when there is a failure.